# Pharse 1(Data Exploration & Initial Assessment)

In [97]:
import pandas as pd
import numpy as np

### Load dataset

In [98]:
a = pd.read_csv("pubg3.csv")

### Observation:
- Dataset has been loaded successfully using pd.read_csv()

### Understanding the Structure of the Dataset

In [99]:
a.shape

(49450, 15)

### Observation:
- The dataset contains 50000 rows and 15 columns representing Gameplay Performance Insights.

### Previewing the first few rows

In [100]:
a.head()

Unnamed: 0  assists  boosts  damageDealt  kills  walkDistance  \
0           0      NaN     1.0          NaN    2.0       2172.00   
1           1      0.0     0.0         0.00    0.0        836.10   
2           2      0.0     0.0         0.00    0.0         36.83   
3           3      0.0     0.0        72.32    0.0        180.20   
4           4      0.0     0.0         0.00    0.0        142.20   

   swimDistance  weaponsAcquired  revives match_category performance_tag  \
0           0.0              6.0      0.0            NaN         Passive   
1           0.0              NaN      0.0         Silver             NaN   
2           0.0              0.0      0.0            NaN             NaN   
3           0.0              2.0      0.0         Silver         Passive   
4           NaN              1.0      0.0            NaN         Passive   

  distance_type healing_style kill_style  winPlacePerc  
0          High       LowHeal     Normal        0.8333  
1        Medium           NaN        NaN        0.3750  
2           NaN        NoHeal        NaN        0.0000  
3           Low        NoHeal        NaN           NaN  
4           Low           NaN        NaN        0.1522

### Observation:
- The preview shows the first few rows of the data and helps us check if the file loaded correctly.

### Dataset Structure and Data Types

In [101]:
a.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49450 entries, 0 to 49449
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Unnamed: 0       49450 non-null  int64  
 1   assists          42850 non-null  float64
 2   boosts           42011 non-null  float64
 3   damageDealt      45951 non-null  float64
 4   kills            45911 non-null  float64
 5   walkDistance     43886 non-null  float64
 6   swimDistance     41258 non-null  float64
 7   weaponsAcquired  45760 non-null  float64
 8   revives          41204 non-null  float64
 9   match_category   40724 non-null  object 
 10  performance_tag  44961 non-null  object 
 11  distance_type    43218 non-null  object 
 12  healing_style    40842 non-null  object 
 13  kill_style       20176 non-null  object 
 14  winPlacePerc     40720 non-null  float64
dtypes: float64(9), int64(1), object(5)
memory usage: 5.7+ MB


### Observation:
- The dataset contains 50000 records and 15 columns.
- Out of the 15 columns:
- 5 are categorical (object type), such as match_category, performance_tag, and distance_type etc...
- 10 are numerical (float64), including assists and boosts etc...
- Mostly all columns have missing values like assists, damageDealt, rideDistance, and distance_type etc..
- The dataset occupies approximately 5.7+ MB of memory.

### Statistical Overview of Numerical Data

In [102]:
a.describe()

Unnamed: 0  assists        boosts   damageDealt         kills  \
count  49450.000000  42850.0  42011.000000  45951.000000  45911.000000   
mean   24970.560324      0.0      1.063293    123.175613      0.721908   
std    14436.650794      0.0      1.520413    134.568802      0.926770   
min        0.000000      0.0      0.000000      0.000000      0.000000   
25%    12453.250000      0.0      0.000000      0.000000      0.000000   
50%    24963.500000      0.0      0.000000     88.240000      0.000000   
75%    37471.750000      0.0      2.000000    188.800000      1.000000   
max    49999.000000      0.0      5.000000    472.000000      2.500000   

       walkDistance  swimDistance  weaponsAcquired  revives  winPlacePerc  
count  43886.000000       41258.0     45760.000000  41204.0  40720.000000  
mean    1164.870132           0.0         3.659594      0.0      0.478296  
std     1164.926485           0.0         2.217869      0.0      0.304776  
min        0.000000           0.0         0.000000      0.0      0.000000  
25%      166.300000           0.0         2.000000      0.0      0.208300  
50%      720.400000           0.0         3.000000      0.0      0.463900  
75%     1994.000000           0.0         5.000000      0.0      0.744700  
max     4735.550000           0.0         9.500000      0.0      1.000000

### Observation:
- The summary shows fundamental statistical measures like mean, min, max, and standard deviation for all numerical columns.
- For example: The average Mean walkDistance is around 696.900000, with rideDistance ranging from 0.000000 to 18740.000000.
- Some values indicate the presence of outliers.

### Null Value Count Across All Columns

In [103]:
a.isnull().sum()

Unnamed: 0             0
assists             6600
boosts              7439
damageDealt         3499
kills               3539
walkDistance        5564
swimDistance        8192
weaponsAcquired     3690
revives             8246
match_category      8726
performance_tag     4489
distance_type       6232
healing_style       8608
kill_style         29274
winPlacePerc        8730
dtype: int64

### Percentage of Missing Values

In [104]:
round(a.isnull().sum()/len(a)*100,2)

Unnamed: 0          0.00
assists            13.35
boosts             15.04
damageDealt         7.08
kills               7.16
walkDistance       11.25
swimDistance       16.57
weaponsAcquired     7.46
revives            16.68
match_category     17.65
performance_tag     9.08
distance_type      12.60
healing_style      17.41
kill_style         59.20
winPlacePerc       17.65
dtype: float64

Observation:
- The dataset exhibits missing values across multiple columns.
- assists has 13.26% missing values.
- Type of missing value: Missing at Random (MAR)
- boosts has 15.01% missing values.
- Type of missing value: Missing at Random (MAR) 
- damageDealt and Score has 7.02%  missing values.
- Type of missing value: Missing at Random (MAR)
- For all remaining columns like weaponsAcquired,revives has 7.42% and 16.62% missing values.
- Type of missing value: Missing Completely at Random (MCAR)

In [105]:
a.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49450 entries, 0 to 49449
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Unnamed: 0       49450 non-null  int64  
 1   assists          42850 non-null  float64
 2   boosts           42011 non-null  float64
 3   damageDealt      45951 non-null  float64
 4   kills            45911 non-null  float64
 5   walkDistance     43886 non-null  float64
 6   swimDistance     41258 non-null  float64
 7   weaponsAcquired  45760 non-null  float64
 8   revives          41204 non-null  float64
 9   match_category   40724 non-null  object 
 10  performance_tag  44961 non-null  object 
 11  distance_type    43218 non-null  object 
 12  healing_style    40842 non-null  object 
 13  kill_style       20176 non-null  object 
 14  winPlacePerc     40720 non-null  float64
dtypes: float64(9), int64(1), object(5)
memory usage: 5.7+ MB


### Value Count Summary (Gameplay Performance Insights)
#### 1. assists

- Number of assists by a player in a match.

- Type: Float

- Range: 0 – (varies per match).

- Example values: 0, 1, 2, 3.

#### 2. boosts

- Number of boost items (like energy drinks, painkillers) used by the player.

- Type: Float

- Range: 0 – (varies).

- Example values: 0, 1, 3, 5.

#### 3. damageDealt

- Total damage dealt to opponents.

- Type: Float

- Range: 0.0 – (varies).

- Example values: 0.0, 72.3, 250.5, 1000+.

#### 4. kills

- Total number of opponents killed.

- Type: Float

- Range: 0 – (varies).

- Example values: 0, 1, 2, 5.

#### 5. walkDistance

- Total distance walked (in meters).

- Type: Float

- Example values: 36.8, 180.2, 2172.0.

#### 6. rideDistance

- Distance traveled by vehicles.

- Type: Float

- Example values: 0.0, 473.3, 2817.0.

#### 7. swimDistance

- Distance traveled swimming.

- Type: Float

- Example values: 0.0, 10.5, 200.0.

#### 8. weaponsAcquired

- Number of weapons picked up.

- Type: Float

- Example values: 0, 1, 2, 6.

#### 9. revives

- Number of teammates revived.

- Type: Float

- Example values: 0, 1, 2.

#### 10. match_category

- Match ranking category (e.g., Bronze, Silver, Gold).

- Type: Object (Categorical).

- Example values: Silver, Gold.

#### 11. performance_tag

- Overall player performance label.

- Type: Object (Categorical).

- Example values: Passive, Aggressive.

#### 12. distance_type

- Classification of player’s movement distance (e.g., Low, Medium, High).

- Type: Object (Categorical).

- Example values: Low, Medium, High.

#### 13. healing_style

- Healing behavior/style of player.

- Type: Object (Categorical).

- Example values: LowHeal, NoHeal, FullHeal.

#### 14. kill_style

- Style of player’s kills.

- Type: Object (Categorical).

- Example values: Normal, Sniper, CloseCombat.

#### 15. winPlacePerc

- Percentage ranking of the player in the match (scaled between 0 and 1).

- Type: Float

- Example values: 0.0000, 0.1522, 0.3750, 0.8333.

In [106]:
a['assists'].value_counts()

assists
0.0    42850
Name: count, dtype: int64

In [107]:
a['boosts'].value_counts()

boosts
0.0    23583
1.0     6416
2.0     4737
3.0     2863
5.0     2543
4.0     1869
Name: count, dtype: int64

In [108]:
a['damageDealt'].value_counts()

damageDealt
0.00      12453
100.00     2708
472.00     2226
200.00      695
75.00       134
          ...  
460.10        1
43.14         1
27.97         1
437.50        1
65.40         1
Name: count, Length: 8530, dtype: int64

In [109]:
a['kills'].value_counts()

kills
0.0    25849
1.0     9682
2.5     5403
2.0     4977
Name: count, dtype: int64

In [110]:
a['walkDistance'].value_counts()

walkDistance
0.00       529
4735.55    272
1033.00     19
1456.00     19
1553.00     19
          ... 
59.57        1
61.60        1
685.30       1
754.10       1
41.29        1
Name: count, Length: 15534, dtype: int64

In [111]:
a['swimDistance'].value_counts()

swimDistance
0.0    41258
Name: count, dtype: int64

In [112]:
a['weaponsAcquired'].value_counts()

weaponsAcquired
3.0    8038
2.0    8004
4.0    7247
1.0    5898
5.0    5636
6.0    3759
7.0    2385
0.0    1912
8.0    1320
9.5     834
9.0     727
Name: count, dtype: int64

In [113]:
a['revives'].value_counts()

revives
0.0    41204
Name: count, dtype: int64

In [114]:
a['winPlacePerc'].value_counts()

winPlacePerc
0.0000    1585
1.0000    1155
0.5000     496
0.3333     402
0.6667     342
          ... 
0.1414       1
0.8537       1
0.6866       1
0.9780       1
0.8384       1
Name: count, Length: 1369, dtype: int64

### Observations (Gameplay Performance Insights – Float Columns)
#### 1. assists

- Number of times a player assisted in kills.

- Each row means:
0.0 → 35,790 → 35,790 players had 0 assists.
1.0 → 5,722 → 5,722 players had 1 assist.
2.0 → 1,358 → 1,358 players had 2 assists.
3.0 → 348 → 348 players had 3 assists.
4.0 → 91 → 91 players had 4 assists.

#### 2. boosts

- Number of boost items used (e.g., energy drinks, painkillers).

- Each row means:
0.0 → 24,066 → 24,066 players used no boosts.
1.0 → 6,416 → 6,416 players used 1 boost.
2.0 → 4,738 → 4,738 players used 2 boosts.
3.0 → 2,863 → 2,863 players used 3 boosts.
4.0 → 1,869 → 1,869 players used 4 boosts.

#### 3. damageDealt

- Total damage dealt by a player.

- Each row means:
0.0 → 12,986 → 12,986 players dealt no damage.
100.0 → 2,708 → 2,708 players dealt 100 damage.
200.0 → 695 → 695 players dealt 200 damage.
75.0 → 134 → 134 players dealt 75 damage.
300.0 → 134 → 134 players dealt 300 damage.

#### 4. kills

- Number of kills by a player.

- Each row means:
0.0 → 26,381 → 26,381 players had no kills.
1.0 → 9,683 → 9,683 players had 1 kill.
2.0 → 4,978 → 4,978 players had 2 kills.
3.0 → 2,442 → 2,442 players had 3 kills.
4.0 → 1,279 → 1,279 players had 4 kills.

#### 5. walkDistance

- Distance walked by players (meters).

- Each row means:
0.0 → 963 → 963 players did not walk.
1456.0 → 19 → 19 players walked 1456m.
1553.0 → 19 → 19 players walked 1553m.
1033.0 → 19 → 19 players walked 1033m.
1689.0 → 18 → 18 players walked 1689m.

#### 6. rideDistance

- Distance traveled in vehicles (meters).

- Each row means:
0.0 → 31,112 → 31,112 players did not use vehicles.
1634.0 → 9 → 9 players rode 1634m.
1238.0 → 8 → 8 players rode 1238m.
3856.0 → 8 → 8 players rode 3856m.
1096.0 → 8 → 8 players rode 1096m.

#### 7. swimDistance

- Distance swum by players (meters).

- Each row means:
0.0 → 39,085 → 39,085 players did not swim.
124.7 → 5 → 5 players swam 124.7m.
10.81 → 5 → 5 players swam 10.81m.
105.3 → 4 → 4 players swam 105.3m.
13.09 → 4 → 4 players swam 13.09m.

#### 8. weaponsAcquired

- Number of weapons picked up.

- Each row means:
3.0 → 8,042 → 8,042 players picked up 3 weapons.
2.0 → 8,015 → 8,015 players picked up 2 weapons.
4.0 → 7,249 → 7,249 players picked up 4 weapons.
1.0 → 5,946 → 5,946 players picked up 1 weapon.
5.0 → 5,637 → 5,637 players picked up 5 weapons.

#### 9. revives

- Number of teammates revived.

- Each row means:
0.0 → 36,126 → 36,126 players did not revive teammates.
1.0 → 4,503 → 4,503 players revived 1 teammate.
2.0 → 855 → 855 players revived 2 teammates.
3.0 → 163 → 163 players revived 3 teammates.
4.0 → 31 → 31 players revived 4 teammates.

#### 10. winPlacePerc

- Player’s win placement percentage (scaled 0 to 1).

- Each row means:
0.0 → 2,005 → 2,005 players ranked last (0%).
1.0 → 1,155 → 1,155 players ranked first (100%).
0.5 → 496 → 496 players ranked 50% percentile.
0.3333 → 402 → 402 players ranked 33% percentile.
0.6667 → 342 → 342 players ranked 67% percentile.

In [115]:
a.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49450 entries, 0 to 49449
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Unnamed: 0       49450 non-null  int64  
 1   assists          42850 non-null  float64
 2   boosts           42011 non-null  float64
 3   damageDealt      45951 non-null  float64
 4   kills            45911 non-null  float64
 5   walkDistance     43886 non-null  float64
 6   swimDistance     41258 non-null  float64
 7   weaponsAcquired  45760 non-null  float64
 8   revives          41204 non-null  float64
 9   match_category   40724 non-null  object 
 10  performance_tag  44961 non-null  object 
 11  distance_type    43218 non-null  object 
 12  healing_style    40842 non-null  object 
 13  kill_style       20176 non-null  object 
 14  winPlacePerc     40720 non-null  float64
dtypes: float64(9), int64(1), object(5)
memory usage: 5.7+ MB


In [116]:
a['match_category'].value_counts()

match_category
Platinum    10342
Gold        10296
Silver      10280
Bronze       9806
Name: count, dtype: int64

In [117]:
a['performance_tag'].value_counts()

performance_tag
Passive       43961
Aggressive     1000
Name: count, dtype: int64

In [118]:
a['distance_type'].value_counts()

distance_type
Medium     13504
Low        11420
High       10771
VeryLow     7523
Name: count, dtype: int64

In [119]:
a['healing_style'].value_counts()

healing_style
NoHeal      24039
LowHeal     13749
HighHeal     3054
Name: count, dtype: int64

In [120]:
a['kill_style'].value_counts()

kill_style
Normal      12284
Headshot     7892
Name: count, dtype: int64

### Observations (Gameplay Performance Insights – Object Columns)
#### 1. match_category

- The rank category a player’s match falls into (like Bronze, Silver, Gold).

- Each row means:
- Silver → Many players’ matches fall into the Silver category.
- Gold → Many matches are categorized as Gold.

- Most categories occur multiple times, but some may be rarer (e.g., higher-tier categories).

#### 2. performance_tag

- Overall performance label of a player.

- Each row means:
- Passive → Appears frequently, showing many players take a cautious style.
- Aggressive → Appears often for high-engagement players.

- Only a few distinct tags exist, and they repeat often.

#### 3. distance_type

- Classification of total distance traveled by a player.

- Each row means:
- Low → Many players traveled very little distance.
- Medium → A significant portion traveled a moderate distance.
- High → Fewer players reached very high distances.

- Categories are limited and descriptive rather than unique.

#### 4. healing_style

- The way players approached healing during matches.

- Each row means:
- NoHeal → Players didn’t use healing items.
- LowHeal → Minimal healing use.
- FullHeal → Heavy reliance on healing.

- Categories are broad, covering healing strategies.

#### 5. kill_style

- Describes how a player achieved kills.

- Each row means:
- Normal → Most common style of killing opponents.
- Sniper / CloseCombat → Appear less frequently depending on playstyle.

- Distribution shows different tactical approaches.

### Checking for Duplicates

In [121]:
a.duplicated().sum()

0

### Observation:
 - As seen there are 550 duplicates found in the dataset.

In [122]:
a.drop_duplicates(inplace=True)

In [123]:
a.duplicated().sum()

0

# outlier detection

In [124]:
def print_no_of_outliers(x):
    q1 = a[x].quantile(0.25)
    q3 = a[x].quantile(0.75)
    iqr = q3 - q1
    lb = q1 - 1.5 * iqr
    ub = q3 + 1.5 * iqr
    length = len(a[(a[x] < lb) | (a[x] > ub)])
    print(length)

In [125]:
print_no_of_outliers("assists")

0


In [126]:
print_no_of_outliers("boosts")

0


In [127]:
print_no_of_outliers("damageDealt")

0


In [128]:
print_no_of_outliers("kills")

0


In [129]:
print_no_of_outliers("walkDistance")

0


In [131]:
print_no_of_outliers("swimDistance")

0


In [132]:
print_no_of_outliers("weaponsAcquired")

0


In [133]:
print_no_of_outliers("revives")

0


In [134]:
print_no_of_outliers("winPlacePerc")

0


# outlier replacement

In [136]:
def cap_outliers_iqr(df,column):
    Q1 = np.percentile(df[column].dropna(), 25)
    Q3 = np.percentile(df[column].dropna(), 75)
    IQR = Q3 - Q1
    lower_bound = Q1 - 1.5 * IQR
    upper_bound = Q3 + 1.5 * IQR
    df.loc[df[column] < lower_bound, column] = lower_bound
    df.loc[df[column] > upper_bound, column] = upper_bound
    return df

In [137]:
cap_outliers_iqr(a,"assists")

Unnamed: 0  assists  boosts  damageDealt  kills  walkDistance  \
0               0      NaN     1.0          NaN    2.0       2172.00   
1               1      0.0     0.0         0.00    0.0        836.10   
2               2      0.0     0.0         0.00    0.0         36.83   
3               3      0.0     0.0        72.32    0.0        180.20   
4               4      0.0     0.0         0.00    0.0        142.20   
...           ...      ...     ...          ...    ...           ...   
49445       49995      0.0     0.0          NaN    0.0         41.29   
49446       49996      0.0     NaN       100.00    1.0       2504.00   
49447       49997      0.0     NaN          NaN    2.0       2421.00   
49448       49998      0.0     0.0       175.00    2.0        266.10   
49449       49999      0.0     0.0         0.00    0.0        387.40   

       swimDistance  weaponsAcquired  revives match_category performance_tag  \
0               0.0              6.0      0.0            NaN         Passive   
1               0.0              NaN      0.0         Silver             NaN   
2               0.0              0.0      0.0            NaN             NaN   
3               0.0              2.0      0.0         Silver         Passive   
4               NaN              1.0      0.0            NaN         Passive   
...             ...              ...      ...            ...             ...   
49445           0.0              3.0      NaN         Silver         Passive   
49446           0.0              2.0      0.0           Gold         Passive   
49447           0.0              5.0      0.0            NaN         Passive   
49448           0.0              NaN      0.0           Gold         Passive   
49449           0.0              3.0      0.0           Gold         Passive   

      distance_type healing_style kill_style  winPlacePerc  
0              High       LowHeal     Normal        0.8333  
1            Medium           NaN        NaN        0.3750  
2               NaN        NoHeal        NaN        0.0000  
3               Low        NoHeal        NaN           NaN  
4               Low           NaN        NaN        0.1522  
...             ...           ...        ...           ...  
49445       VeryLow           NaN        NaN        0.2143  
49446          High       LowHeal     Normal           NaN  
49447          High       LowHeal     Normal        0.9655  
49448           Low       LowHeal     Normal           NaN  
49449           Low        NoHeal        NaN           NaN  

[49450 rows x 15 columns]

In [138]:
cap_outliers_iqr(a,"boosts")

Unnamed: 0  assists  boosts  damageDealt  kills  walkDistance  \
0               0      NaN     1.0          NaN    2.0       2172.00   
1               1      0.0     0.0         0.00    0.0        836.10   
2               2      0.0     0.0         0.00    0.0         36.83   
3               3      0.0     0.0        72.32    0.0        180.20   
4               4      0.0     0.0         0.00    0.0        142.20   
...           ...      ...     ...          ...    ...           ...   
49445       49995      0.0     0.0          NaN    0.0         41.29   
49446       49996      0.0     NaN       100.00    1.0       2504.00   
49447       49997      0.0     NaN          NaN    2.0       2421.00   
49448       49998      0.0     0.0       175.00    2.0        266.10   
49449       49999      0.0     0.0         0.00    0.0        387.40   

       swimDistance  weaponsAcquired  revives match_category performance_tag  \
0               0.0              6.0      0.0            NaN         Passive   
1               0.0              NaN      0.0         Silver             NaN   
2               0.0              0.0      0.0            NaN             NaN   
3               0.0              2.0      0.0         Silver         Passive   
4               NaN              1.0      0.0            NaN         Passive   
...             ...              ...      ...            ...             ...   
49445           0.0              3.0      NaN         Silver         Passive   
49446           0.0              2.0      0.0           Gold         Passive   
49447           0.0              5.0      0.0            NaN         Passive   
49448           0.0              NaN      0.0           Gold         Passive   
49449           0.0              3.0      0.0           Gold         Passive   

      distance_type healing_style kill_style  winPlacePerc  
0              High       LowHeal     Normal        0.8333  
1            Medium           NaN        NaN        0.3750  
2               NaN        NoHeal        NaN        0.0000  
3               Low        NoHeal        NaN           NaN  
4               Low           NaN        NaN        0.1522  
...             ...           ...        ...           ...  
49445       VeryLow           NaN        NaN        0.2143  
49446          High       LowHeal     Normal           NaN  
49447          High       LowHeal     Normal        0.9655  
49448           Low       LowHeal     Normal           NaN  
49449           Low        NoHeal        NaN           NaN  

[49450 rows x 15 columns]

In [139]:
cap_outliers_iqr(a,"damageDealt")

Unnamed: 0  assists  boosts  damageDealt  kills  walkDistance  \
0               0      NaN     1.0          NaN    2.0       2172.00   
1               1      0.0     0.0         0.00    0.0        836.10   
2               2      0.0     0.0         0.00    0.0         36.83   
3               3      0.0     0.0        72.32    0.0        180.20   
4               4      0.0     0.0         0.00    0.0        142.20   
...           ...      ...     ...          ...    ...           ...   
49445       49995      0.0     0.0          NaN    0.0         41.29   
49446       49996      0.0     NaN       100.00    1.0       2504.00   
49447       49997      0.0     NaN          NaN    2.0       2421.00   
49448       49998      0.0     0.0       175.00    2.0        266.10   
49449       49999      0.0     0.0         0.00    0.0        387.40   

       swimDistance  weaponsAcquired  revives match_category performance_tag  \
0               0.0              6.0      0.0            NaN         Passive   
1               0.0              NaN      0.0         Silver             NaN   
2               0.0              0.0      0.0            NaN             NaN   
3               0.0              2.0      0.0         Silver         Passive   
4               NaN              1.0      0.0            NaN         Passive   
...             ...              ...      ...            ...             ...   
49445           0.0              3.0      NaN         Silver         Passive   
49446           0.0              2.0      0.0           Gold         Passive   
49447           0.0              5.0      0.0            NaN         Passive   
49448           0.0              NaN      0.0           Gold         Passive   
49449           0.0              3.0      0.0           Gold         Passive   

      distance_type healing_style kill_style  winPlacePerc  
0              High       LowHeal     Normal        0.8333  
1            Medium           NaN        NaN        0.3750  
2               NaN        NoHeal        NaN        0.0000  
3               Low        NoHeal        NaN           NaN  
4               Low           NaN        NaN        0.1522  
...             ...           ...        ...           ...  
49445       VeryLow           NaN        NaN        0.2143  
49446          High       LowHeal     Normal           NaN  
49447          High       LowHeal     Normal        0.9655  
49448           Low       LowHeal     Normal           NaN  
49449           Low        NoHeal        NaN           NaN  

[49450 rows x 15 columns]

In [140]:
cap_outliers_iqr(a,"kills")

Unnamed: 0  assists  boosts  damageDealt  kills  walkDistance  \
0               0      NaN     1.0          NaN    2.0       2172.00   
1               1      0.0     0.0         0.00    0.0        836.10   
2               2      0.0     0.0         0.00    0.0         36.83   
3               3      0.0     0.0        72.32    0.0        180.20   
4               4      0.0     0.0         0.00    0.0        142.20   
...           ...      ...     ...          ...    ...           ...   
49445       49995      0.0     0.0          NaN    0.0         41.29   
49446       49996      0.0     NaN       100.00    1.0       2504.00   
49447       49997      0.0     NaN          NaN    2.0       2421.00   
49448       49998      0.0     0.0       175.00    2.0        266.10   
49449       49999      0.0     0.0         0.00    0.0        387.40   

       swimDistance  weaponsAcquired  revives match_category performance_tag  \
0               0.0              6.0      0.0            NaN         Passive   
1               0.0              NaN      0.0         Silver             NaN   
2               0.0              0.0      0.0            NaN             NaN   
3               0.0              2.0      0.0         Silver         Passive   
4               NaN              1.0      0.0            NaN         Passive   
...             ...              ...      ...            ...             ...   
49445           0.0              3.0      NaN         Silver         Passive   
49446           0.0              2.0      0.0           Gold         Passive   
49447           0.0              5.0      0.0            NaN         Passive   
49448           0.0              NaN      0.0           Gold         Passive   
49449           0.0              3.0      0.0           Gold         Passive   

      distance_type healing_style kill_style  winPlacePerc  
0              High       LowHeal     Normal        0.8333  
1            Medium           NaN        NaN        0.3750  
2               NaN        NoHeal        NaN        0.0000  
3               Low        NoHeal        NaN           NaN  
4               Low           NaN        NaN        0.1522  
...             ...           ...        ...           ...  
49445       VeryLow           NaN        NaN        0.2143  
49446          High       LowHeal     Normal           NaN  
49447          High       LowHeal     Normal        0.9655  
49448           Low       LowHeal     Normal           NaN  
49449           Low        NoHeal        NaN           NaN  

[49450 rows x 15 columns]

In [141]:
cap_outliers_iqr(a,"walkDistance")

Unnamed: 0  assists  boosts  damageDealt  kills  walkDistance  \
0               0      NaN     1.0          NaN    2.0       2172.00   
1               1      0.0     0.0         0.00    0.0        836.10   
2               2      0.0     0.0         0.00    0.0         36.83   
3               3      0.0     0.0        72.32    0.0        180.20   
4               4      0.0     0.0         0.00    0.0        142.20   
...           ...      ...     ...          ...    ...           ...   
49445       49995      0.0     0.0          NaN    0.0         41.29   
49446       49996      0.0     NaN       100.00    1.0       2504.00   
49447       49997      0.0     NaN          NaN    2.0       2421.00   
49448       49998      0.0     0.0       175.00    2.0        266.10   
49449       49999      0.0     0.0         0.00    0.0        387.40   

       swimDistance  weaponsAcquired  revives match_category performance_tag  \
0               0.0              6.0      0.0            NaN         Passive   
1               0.0              NaN      0.0         Silver             NaN   
2               0.0              0.0      0.0            NaN             NaN   
3               0.0              2.0      0.0         Silver         Passive   
4               NaN              1.0      0.0            NaN         Passive   
...             ...              ...      ...            ...             ...   
49445           0.0              3.0      NaN         Silver         Passive   
49446           0.0              2.0      0.0           Gold         Passive   
49447           0.0              5.0      0.0            NaN         Passive   
49448           0.0              NaN      0.0           Gold         Passive   
49449           0.0              3.0      0.0           Gold         Passive   

      distance_type healing_style kill_style  winPlacePerc  
0              High       LowHeal     Normal        0.8333  
1            Medium           NaN        NaN        0.3750  
2               NaN        NoHeal        NaN        0.0000  
3               Low        NoHeal        NaN           NaN  
4               Low           NaN        NaN        0.1522  
...             ...           ...        ...           ...  
49445       VeryLow           NaN        NaN        0.2143  
49446          High       LowHeal     Normal           NaN  
49447          High       LowHeal     Normal        0.9655  
49448           Low       LowHeal     Normal           NaN  
49449           Low        NoHeal        NaN           NaN  

[49450 rows x 15 columns]

In [142]:
cap_outliers_iqr(a,"swimDistance")

Unnamed: 0  assists  boosts  damageDealt  kills  walkDistance  \
0               0      NaN     1.0          NaN    2.0       2172.00   
1               1      0.0     0.0         0.00    0.0        836.10   
2               2      0.0     0.0         0.00    0.0         36.83   
3               3      0.0     0.0        72.32    0.0        180.20   
4               4      0.0     0.0         0.00    0.0        142.20   
...           ...      ...     ...          ...    ...           ...   
49445       49995      0.0     0.0          NaN    0.0         41.29   
49446       49996      0.0     NaN       100.00    1.0       2504.00   
49447       49997      0.0     NaN          NaN    2.0       2421.00   
49448       49998      0.0     0.0       175.00    2.0        266.10   
49449       49999      0.0     0.0         0.00    0.0        387.40   

       swimDistance  weaponsAcquired  revives match_category performance_tag  \
0               0.0              6.0      0.0            NaN         Passive   
1               0.0              NaN      0.0         Silver             NaN   
2               0.0              0.0      0.0            NaN             NaN   
3               0.0              2.0      0.0         Silver         Passive   
4               NaN              1.0      0.0            NaN         Passive   
...             ...              ...      ...            ...             ...   
49445           0.0              3.0      NaN         Silver         Passive   
49446           0.0              2.0      0.0           Gold         Passive   
49447           0.0              5.0      0.0            NaN         Passive   
49448           0.0              NaN      0.0           Gold         Passive   
49449           0.0              3.0      0.0           Gold         Passive   

      distance_type healing_style kill_style  winPlacePerc  
0              High       LowHeal     Normal        0.8333  
1            Medium           NaN        NaN        0.3750  
2               NaN        NoHeal        NaN        0.0000  
3               Low        NoHeal        NaN           NaN  
4               Low           NaN        NaN        0.1522  
...             ...           ...        ...           ...  
49445       VeryLow           NaN        NaN        0.2143  
49446          High       LowHeal     Normal           NaN  
49447          High       LowHeal     Normal        0.9655  
49448           Low       LowHeal     Normal           NaN  
49449           Low        NoHeal        NaN           NaN  

[49450 rows x 15 columns]

In [143]:
cap_outliers_iqr(a,"weaponsAcquired")

Unnamed: 0  assists  boosts  damageDealt  kills  walkDistance  \
0               0      NaN     1.0          NaN    2.0       2172.00   
1               1      0.0     0.0         0.00    0.0        836.10   
2               2      0.0     0.0         0.00    0.0         36.83   
3               3      0.0     0.0        72.32    0.0        180.20   
4               4      0.0     0.0         0.00    0.0        142.20   
...           ...      ...     ...          ...    ...           ...   
49445       49995      0.0     0.0          NaN    0.0         41.29   
49446       49996      0.0     NaN       100.00    1.0       2504.00   
49447       49997      0.0     NaN          NaN    2.0       2421.00   
49448       49998      0.0     0.0       175.00    2.0        266.10   
49449       49999      0.0     0.0         0.00    0.0        387.40   

       swimDistance  weaponsAcquired  revives match_category performance_tag  \
0               0.0              6.0      0.0            NaN         Passive   
1               0.0              NaN      0.0         Silver             NaN   
2               0.0              0.0      0.0            NaN             NaN   
3               0.0              2.0      0.0         Silver         Passive   
4               NaN              1.0      0.0            NaN         Passive   
...             ...              ...      ...            ...             ...   
49445           0.0              3.0      NaN         Silver         Passive   
49446           0.0              2.0      0.0           Gold         Passive   
49447           0.0              5.0      0.0            NaN         Passive   
49448           0.0              NaN      0.0           Gold         Passive   
49449           0.0              3.0      0.0           Gold         Passive   

      distance_type healing_style kill_style  winPlacePerc  
0              High       LowHeal     Normal        0.8333  
1            Medium           NaN        NaN        0.3750  
2               NaN        NoHeal        NaN        0.0000  
3               Low        NoHeal        NaN           NaN  
4               Low           NaN        NaN        0.1522  
...             ...           ...        ...           ...  
49445       VeryLow           NaN        NaN        0.2143  
49446          High       LowHeal     Normal           NaN  
49447          High       LowHeal     Normal        0.9655  
49448           Low       LowHeal     Normal           NaN  
49449           Low        NoHeal        NaN           NaN  

[49450 rows x 15 columns]

In [144]:
cap_outliers_iqr(a,"revives")

Unnamed: 0  assists  boosts  damageDealt  kills  walkDistance  \
0               0      NaN     1.0          NaN    2.0       2172.00   
1               1      0.0     0.0         0.00    0.0        836.10   
2               2      0.0     0.0         0.00    0.0         36.83   
3               3      0.0     0.0        72.32    0.0        180.20   
4               4      0.0     0.0         0.00    0.0        142.20   
...           ...      ...     ...          ...    ...           ...   
49445       49995      0.0     0.0          NaN    0.0         41.29   
49446       49996      0.0     NaN       100.00    1.0       2504.00   
49447       49997      0.0     NaN          NaN    2.0       2421.00   
49448       49998      0.0     0.0       175.00    2.0        266.10   
49449       49999      0.0     0.0         0.00    0.0        387.40   

       swimDistance  weaponsAcquired  revives match_category performance_tag  \
0               0.0              6.0      0.0            NaN         Passive   
1               0.0              NaN      0.0         Silver             NaN   
2               0.0              0.0      0.0            NaN             NaN   
3               0.0              2.0      0.0         Silver         Passive   
4               NaN              1.0      0.0            NaN         Passive   
...             ...              ...      ...            ...             ...   
49445           0.0              3.0      NaN         Silver         Passive   
49446           0.0              2.0      0.0           Gold         Passive   
49447           0.0              5.0      0.0            NaN         Passive   
49448           0.0              NaN      0.0           Gold         Passive   
49449           0.0              3.0      0.0           Gold         Passive   

      distance_type healing_style kill_style  winPlacePerc  
0              High       LowHeal     Normal        0.8333  
1            Medium           NaN        NaN        0.3750  
2               NaN        NoHeal        NaN        0.0000  
3               Low        NoHeal        NaN           NaN  
4               Low           NaN        NaN        0.1522  
...             ...           ...        ...           ...  
49445       VeryLow           NaN        NaN        0.2143  
49446          High       LowHeal     Normal           NaN  
49447          High       LowHeal     Normal        0.9655  
49448           Low       LowHeal     Normal           NaN  
49449           Low        NoHeal        NaN           NaN  

[49450 rows x 15 columns]

In [145]:
a.to_csv("pubg3.csv")